lib/fdk/runner.rb

```diff
@@ -8,19 +8,43 @@
 module FDK
   def self.handle(func)
     format = ENV['FN_FORMAT']
-    if format == 'json'
+    if format == 'cloudevent'
       parser = Yajl::Parser.new
 
-      parser.on_parse_complete = lambda do |payload|
-        context = Context.new(payload)
-        body = payload['body']
+      parser.on_parse_complete = lambda do |event|
+        context = Context.new(event)
+        body = event['data']
+        # Skipping json parsing of body because it would already be a parsed map according to the format spec defined here: https://github.com/cloudevents/spec/blob/master/serialization.md#json
+        se = FDK.single_event(function: func, context: context, input: body)
+
+        # Respond with modified event
+        event['data'] = se
+        event['extensions']['protocol'] = {
+          headers: {
+              'Content-Type' => ['application/json']
+          },
+          'status_code' => 200,
```

Comment:
need to find a way to make these more flexible in general going forward (i've some ideas), can discuss on 2.0 doc, this is OK for a time